File: GitToolBox/src/main/java/zielu/gittoolbox/formatter/Formatted.java
package zielu.gittoolbox.formatter;

import static org.apache.commons.lang3.builder.ToStringStyle.SHORT_PREFIX_STYLE;

import org.apache.commons.lang3.StringUtils;
import org.apache.commons.lang3.builder.ToStringBuilder;

public final class Formatted {
  private final String text;
  private final boolean matches;

  Formatted(String text, boolean matches) {
    this.text = text;
    this.matches = matches;
  }

  public String getText() {
    return text;
  }

  public boolean matches() {
    return matches;
  }

  public boolean isDisplayable() {
    return matches && StringUtils.isNotBlank(text);
  }

  @Override
  public String toString() {
    return new ToStringBuilder(this, SHORT_PREFIX_STYLE)
        .append("text", "'" + text + "'")
        .append("matches", matches)
        .toString();
  }
}

File: src/main/kotlin/zielu/gittoolbox/formatter/Formatted.kt
package zielu.gittoolbox.formatter

import org.apache.commons.lang3.StringUtils

internal data class Formatted(
  val text: String,
  val matches: Boolean,
  val message: String? = null
) {
  fun getDisplayable() = matches && StringUtils.isNotBlank(text)
}
